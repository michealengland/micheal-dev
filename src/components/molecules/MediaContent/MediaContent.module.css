.mediaContent {
  @apply text-primary-purple1 items-center;

  & .title {
    @apply text-primary-purple2  text-4xl lg:text-5xl font-normal mb-4;
  }

  & p {
    @apply mb-4;
  }

  & ul {
    @apply leading-relaxed list-disc list-inside m-0;

    & li {
      @apply mb-1;

      &:last-child {
        @apply mb-0;
      }
    }
  }

  @media screen and (min-width: 48em) {
    @apply grid;

    grid-template-areas: 'media-text-content media-text-media';
    grid-template-columns: auto 50%;
  }

  & .content {
    @apply p-8 md:pl-16 lg:pl-40;

    grid-area: media-text-content;
    word-break: break-word;
  }

  & .media {
    @apply block bg-primary-purple1 h-full m-0;

    clip-path: polygon(0 0, 100% 0%, 100% 100%, 0 100%);
    grid-area: media-text-media;

    @screen md {
      &.clipLeft {
        clip-path: polygon(25% 0%, 100% 0%, 100% 100%, 0% 100%);
      }
      &.clipRight {
        clip-path: polygon(0 0, 100% 0%, 100% 100%, 25% 100%);
      }
    }

    & img {
      @apply block h-full m-0 object-cover w-full;
    }
  }
}
