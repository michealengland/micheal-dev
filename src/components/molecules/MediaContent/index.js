import React from 'react'
import PropTypes from 'prop-types'
import Heading from '@components/atoms/Heading'
import createMarkup from '@utilities/createMarkup'

export default function MediaContent({content, img, imgAlt, title, titleTag}) {
  return (
    <section>
      {(title || content) &&
        <div>
          {title && <Heading tag={titleTag}>{title}</Heading>}
          {content && <div dangerouslySetInnerHTML={ createMarkup(content) } />}
        </div>
      }
      {img &&
        <figure>
          <img src={img} alt={imgAlt} />
        </figure>
      }
    </section>
  )
}

MediaContent.propTypes = {
  content: PropTypes.string,
  img: PropTypes.string,
  imgAlt: PropTypes.string,
  title: PropTypes.string,
  titleTag: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6'])
}

MediaContent.defaultProps = {
  imgAlt: '',
  titleTag: 'h2'
}
