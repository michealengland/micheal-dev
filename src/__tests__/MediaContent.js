import React, {Fragment} from 'react'
import renderer from 'react-test-renderer'
import MediaContent from '@components/molecules/MediaContent'

describe('MediaContent', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Fragment>
          <MediaContent
            content={'MediaContent subtitle.'}
            img={'./images/wcus2018-hall-track-1.jpg'}
            title={'MediaContent title.'}
          />
          <MediaContent
            content={'MediaContent subtitle.'}
            img={'./images/wcus2018-hall-track-1.jpg'}
            imgAlt="img alt content"
            imgClipRight={true}
            title={'MediaContent title.'}
            titleTag="h3"
          />
        </Fragment>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
