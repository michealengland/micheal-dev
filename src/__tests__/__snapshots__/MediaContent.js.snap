// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaContent renders correctly 1`] = `
Array [
  <section
    className="mediaContent"
  >
    <div
      className="content"
    >
      <h2
        className="title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "MediaContent title.",
          }
        }
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "MediaContent subtitle.",
          }
        }
      />
    </div>
    <figure
      className="media clipLeft"
    >
      <img
        alt=""
        src="./images/wcus2018-hall-track-1.jpg"
      />
    </figure>
  </section>,
  <section
    className="mediaContent"
  >
    <div
      className="content"
    >
      <h3
        className="title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "MediaContent title.",
          }
        }
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "MediaContent subtitle.",
          }
        }
      />
    </div>
    <figure
      className="media clipRight"
    >
      <img
        alt="img alt content"
        src="./images/wcus2018-hall-track-1.jpg"
      />
    </figure>
  </section>,
]
`;
